### PR TITLE
Add another way to dynamically generate DAGs to docs

### PR DESCRIPTION
**Pull request by makrushin-evgenii**
This PR created as a result of discussion https://github.com/apache/airflow/discussions/21278

### CODE REVIEW

**Comment by dstandish** [comment]
Hi @makrushin-evgenii 

I like this example.  Taking a look at this, it seems this section on "dynamic dag generation" is getting rather long and starting to look like more of a "how to" than a "best practices".  

I think it might be best if we create a "how to" page called "dynamic dag generation" or perhaps "Config-based dag generation".  Then in the "best practices" page we can keep a more succinct message about what really is the "best practice" we're trying to emphasize in this area, and we can link to the how-to ("for more details on dag generation, see our how-to...").

Would you be interested in takking this on as part of this PR?

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
> Hi @makrushin-evgenii
> 
> I like this example. Taking a look at this, it seems this section on "dynamic dag generation" is getting rather long and starting to look like more of a "how to" than a "best practices".
> 
> I think it might be best if we create a "how to" page called "dynamic dag generation" or perhaps "Config-based dag generation". Then in the "best practices" page we can keep a more succinct message about what really is the "best practice" we're trying to emphasize in this area, and we can link to the how-to ("for more details on dag generation, see our how-to...").
> 
> Would you be interested in takking this on as part of this PR?

Might be a good idea actuallly if you want to do it @makrushin-evgenii :)

**Comment by makrushin-evgenii** [comment]
> Hi @makrushin-evgenii
> 
> I like this example. Taking a look at this, it seems this section on "dynamic dag generation" is getting rather long and starting to look like more of a "how to" than a "best practices".
> 
> I think it might be best if we create a "how to" page called "dynamic dag generation" or perhaps "Config-based dag generation". Then in the "best practices" page we can keep a more succinct message about what really is the "best practice" we're trying to emphasize in this area, and we can link to the how-to ("for more details on dag generation, see our how-to...").
> 
> Would you be interested in takking this on as part of this PR?

You're right. Want to do it this way, but forgot about the "How To" page. I'll make changes in a few days

A few clarifications:
1. Should I move all "dynamic dag generation" section content from "best practies" page to "how to", or just mine? If just mine, should I copy all "dynamic dag generation" section content to "how to"?
2. Can I use a links to external resources, e.g. [Astronomer guide "Dynamically Generating DAGs in Airflow"](https://www.astronomer.io/guides/dynamically-generating-dags)?

**Comment by dstandish** [comment]
I was thinking move all of it but just keep any salient points that really are "best practices".  I think the key point there was really that if you are generating a dag from a config to be mindful of network IO because dags are parsed frequently ( though less so now than in the past).  Everything else is more subjective, but still helpful examples to have in a how-to.  But of course apply your own judgment too about what makes sense.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Could you please rebase this one @makrushin-evgenii if it fails?

File: docs/apache-airflow/howto/dynamic-dag-generation.rst [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+            print_message(config['message'])
+
+        globals()[dag_id] = dynamic_generated_dag()
+
+The code below will generate a DAG for each config: ``dynamic_generated_dag_config1`` and ``dynamic_generated_dag_config2``.
+Each of them can run separately with related configuration
+
+.. warning::
+  Using this practice, pay attention to "late binding" behaviour in Python loops. See `that GitHub disscusion <https://github.com/apache/airflow/discussions/21278#discussioncomment-2103559>`_ for more details
```
Comment (makrushin-evgenii) [inline docs/apache-airflow/howto/dynamic-dag-generation.rst]:
```suggestion
  Using this practice, pay attention to "late binding" behaviour in Python loops. See `that GitHub discussion <https://github.com/apache/airflow/discussions/21278#discussioncomment-2103559>`_ for more details
```

File: docs/apache-airflow/howto/dynamic-dag-generation.rst [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    from datetime import datetime
+    from airflow.decorators import dag, task
+
+    configs = {
+        "config1": {
+            "message": "first DAG will receive this message"
+        },
+        "config2": {
+            "message": "second DAG will receive this message"
+        },
```
Comment (makrushin-evgenii) [inline docs/apache-airflow/howto/dynamic-dag-generation.rst]:
```suggestion
        "config1": {"message": "first DAG will receive this message"},
        "config2": {"message": "second DAG will receive this message"},
```

File: docs/apache-airflow/howto/dynamic-dag-generation.rst [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    my_dir = os.path.dirname(os.path.abspath(__file__))
+    configuration_file_path = os.path.join(my_dir, "config.yaml")
+    with open(configuration_file_path) as yaml_file:
+        configuration = yaml.safe_load(yaml_file)
+    # Configuration dict is available here
+
+
+Dynamic DAGs with ``globals()``
+.......................
```
Comment (makrushin-evgenii) [inline docs/apache-airflow/howto/dynamic-dag-generation.rst]:
```suggestion
Dynamic DAGs with ``globals()``
...............................
```

**Comment by makrushin-evgenii** [comment]
@potiuk, i fixed all failed checks. What else can I do?

**Review by dstandish — APPROVED** [review]

**Pull request merged**